src/gui/dbsettings/DatabaseSettingsWidgetGeneral.cpp

```diff
@@ -50,6 +50,7 @@ void DatabaseSettingsWidgetGeneral::initialize()
         m_ui->historyMaxItemsCheckBox->setChecked(true);
     } else {
         m_ui->historyMaxItemsSpinBox->setValue(Metadata::DefaultHistoryMaxItems);
+        m_ui->historyMaxItemsSpinBox->setEnabled(false);
```

Comment:
Is it easy to test it in GUI tests?

Reply:
doesn't seem useful, this is a very minor UI setting. I also could not replicate the original problem reported of the max history items spin box being enabled even though the checkbox was unchecked.

Reply:
> doesn't seem useful, this is a very minor UI setting. I also could not replicate the original problem reported of the max history items spin box being enabled even though the checkbox was unchecked.

To replicate you could make a new db, disable both settings then close keepassxc and relaunch into the settings. I can replicate 100% of the time on linux at current release.

I could still add the test altho i also think its a bit too much for such a minor bug. I could also see that this code is run in coverage already.
![image](https://user-images.githubusercontent.com/15849761/218336160-7a134d15-ea15-452b-b897-2b04d102f51f.png)

Reply:
Ahhh you have to restart the program, missing detail :-)